sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/ConditionEvaluatesToConstant.cs

```diff
@@ -1556,4 +1556,33 @@ void Second(FooContainer fooContainer)
             }
         }
     }
+
+  public class TestNullCoalescing
```

Comment:
could you add some tests with the null coalescing operator
- inside a `while`
- inside a ternary condition (maybe like in line 1540? if it makes sense...)
- inside boolean expression (see examples Foo1-Foo7 lines 85-140)

Also, please add some tests inside `ConditionEvaluatesToConstant.CSharp7.cs`

Also, please have a look at the test at line 1332 to see if it does what it should (it's also using the null coalescing operator `if (b ?? false)`)